src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -308,10 +308,11 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
                     // Although having both would be ambiguous, use impl version for compat. sake.
                     // To handle that properly resolve() would have to support
                     // something like [`ambi_fn`](<SomeStruct as SomeTrait>::ambi_fn)
-                    if kind.is_none() {
-                        kind = resolve_associated_trait_item(did, item_name, ns, &self.cx);
+                    let kind = kind.or_else(|| {
```

Comment:
i would have just chained this on the previous one, rustfmt will align it well. if not i guess this is fine. no biggie either way

Reply:
Done.